sympy/assumptions/handlers/ntheory.py

```diff
@@ -246,3 +250,30 @@ def Basic(expr, assumptions):
                 return None
             return not _even
         return _integer
+
+
+AskCoprimeHandler = Dispatcher('AskCoprimeHandler')
+
+@AskCoprimeHandler.register(Expr, Expr)
+def Expr_Expr(x, y, *, assumptions):
+    if x == y:
+        return False
+
+    assumps = conjuncts(assumptions)
+    if Q.coprime(x,y) in assumps:
+        return True
+    if ~Q.coprime(x,y) in assumps:
+        return False
+    # detect even number or odd number
+    # other logics are not implemented yet
```

Comment:
We can check if x and y are prime and not equal. Though maybe it's better to save more advanced predicates for the sathandlers. Writing down even simple logic in this style gets tedious and easy to make mistakes, and deduction only works if you explicitly assume what you need.